src/app_service/service/community/service.nim

```diff
@@ -86,6 +86,52 @@ QtObject:
     result.allCommunities = initTable[string, CommunityDto]()
     result.myCommunityRequests = @[]
 
+  proc findChatById(id: string, chats: seq[ChatDto]): ChatDto =
+    for chat in chats:
+      if(chat.id == id):
+        return chat
+
+  proc findById(id: string, chats: seq[Chat]): int =
+    for chat in chats:
+      if(chat.id == id):
+        return 0
+    return -1
+
+  proc handleCommunityUpdates(self: Service, communities: seq[CommunityDto], updatedChats: seq[ChatDto]) =
+    let community = communities[0]
+    if(not self.joinedCommunities.hasKey(community.id)):
+      error "error: community doesn't exist"
+      return
+
+    let prev_community = self.joinedCommunities[community.id]
+
+    # channel was added
+    if(community.chats.len > prev_community.chats.len):
+      for chat in community.chats:
+        if findById(chat.id, prev_community.chats) == -1:
+          let updated_chat = findChatById(community.id&chat.id, updatedChats)
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_CREATED, CommunityChatArgs(communityId: community.id, chat: toChat(updated_chat)))
+
+    # channel was removed
+    elif(community.chats.len < prev_community.chats.len):
+      for prv_chat in prev_community.chats:
+        if findById(prv_chat.id, community.chats) == -1:
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_DELETED, CommunityChatIdArgs(communityId: community.id, chatId: community.id&prv_chat.id))
+
+    # some property has changed
+    else:
+      for chat in community.chats:
+        # id is present
+        if findById(chat.id, prev_community.chats) == -1:
+          error "error: chat not present"
+          return
+        # but something is different
+        if(prev_community.chats.find(chat) == -1):
+          let updated_chat = findChatById(community.id&chat.id, updatedChats)
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_EDITED, CommunityChatArgs(communityId: community.id, chat: toChat(updated_chat)))
```

Comment:
@Khushboo-dev-cpp I think that chat content module should be connected to `SIGNAL_COMMUNITY_CHANNEL_EDITED` signal, cause there are property `ChatDetails` which contains all details about the chat and it's exposed to and used on the side of qml.

Reply:
oh I think you are right, both should be connected chatSection and chatContent in this case

Reply:
its done